### ISSUE TITLE
Uncaught Error: getRowAmongFoldedRowIntersectsScreenRow() called for non-folded screenRow!

### ISSUE THREAD

**Issue by ysakmrkm** [t9md/atom-vim-mode-plus#1003]
[Enter steps to reproduce:]

1. 3 folded line is there, 2nd line to move key down `j`
2. first line is ok. but, at 2nd line, error occurred.

**Atom**: 1.23.1 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.10.5
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package 1.24.3

### Stack Trace

Uncaught Error: getRowAmongFoldedRowIntersectsScreenRow() called for non-folded screenRow!

```
At /Users/[name]/.atom/packages/vim-mode-plus/lib/operation-stack.js:151

Error: getRowAmongFoldedRowIntersectsScreenRow() called for non-folded screenRow!
    at getRowAmongFoldedRowIntersectsScreenRow (/packages/vim-mode-plus/lib/utils.js:1008:11)
    at Object.getFoldStartRowForRow (/packages/vim-mode-plus/lib/utils.js:1019:7)
    at MoveDown.getFoldStartRowForRow (/packages/vim-mode-plus/lib/base.js:312:75)
    at MoveDown.getBufferRow (/packages/vim-mode-plus/lib/motion.js:228:17)
    at /packages/vim-mode-plus/lib/motion.js:232:82
    at /packages/vim-mode-plus/lib/motion.js:107:7
    at MoveDown.countTimes (/packages/vim-mode-plus/lib/base.js:70:7)
    at MoveDown.moveCursorCountTimes (/packages/vim-mode-plus/lib/motion.js:106:10)
    at MoveDown.moveCursor (/packages/vim-mode-plus/lib/motion.js:232:10)
    at MoveDown.moveWithSaveJump (/packages/vim-mode-plus/lib/motion.js:48:10)
    at MoveDown.execute (/packages/vim-mode-plus/lib/motion.js:61:14)
    at OperationStack.execute (/packages/vim-mode-plus/lib/operation-stack.js:187:33)
    at OperationStack.process (/packages/vim-mode-plus/lib/operation-stack.js:180:12)
    at OperationStack.run (/packages/vim-mode-plus/lib/operation-stack.js:96:14)
    at /packages/vim-mode-plus/lib/vim-state.js:67:45)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:388:36)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:621:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:412:28)
    at WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:108:40)
```

### Commands

```
     -0:17.5.0 vim-mode-plus:scroll-half-screen-down (input.hidden-input)
 11x -0:17.2.0 vim-mode-plus:move-up (input.hidden-input)
     -0:16.2.0 intentions:highlight (input.hidden-input)
     -0:15.8.0 vim-mode-plus:move-up (input.hidden-input)
     -0:15.7.0 intentions:highlight (input.hidden-input)
     -0:15.6.0 editor:unfold-current-row (input.hidden-input)
     -0:15.1.0 vim-mode-plus:move-down (input.hidden-input)
     -0:15 intentions:highlight (input.hidden-input)
     -0:14.9.0 editor:unfold-current-row (input.hidden-input)
  2x -0:14.5.0 vim-mode-plus:move-down (input.hidden-input)
     -0:14 vim-mode-plus:move-up (input.hidden-input)
     -0:13.8.0 vim-mode-plus:activate-linewise-visual-mode (input.hidden-input)
     -0:13.6.0 vim-mode-plus:move-down (input.hidden-input)
     -0:13.4.0 vim-mode-plus:yank (input.hidden-input)
     -0:13.2.0 vim-mode-plus:move-to-last-line (input.hidden-input)
  3x -0:13 vim-mode-plus:move-up (input.hidden-input)
```

### Non-Core Packages

```
autocomplete-wordpress-hooks 0.2.3 
bem-tools 0.1.15 
block-selection-mode 0.3.0 
browser-plus 0.0.98 
busy-signal 1.4.3 
compass 0.9.1 
editorconfig 2.2.2 
emmet 2.4.3 
file-icons 2.1.15 
filesize 3.0.0 
git-plus 7.10.0 
highlight-selected 0.13.1 
intentions 1.1.5 
japanese-word-selection 0.4.0 
language-apache 1.8.0 
language-jade 0.7.2 
language-pug 0.0.21 
lines 0.13.1 
linter 2.2.0 
linter-htmlhint 1.4.0 
linter-php 1.5.1 
linter-phpcs 1.6.8 
linter-phpmd 2.0.0 
linter-scss-lint 3.1.1 
linter-ui-default 1.6.10 
minimap 4.29.7 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-highlight-selected 4.6.1 
monokai 0.24.0 
move-cursor-improved 0.3.4 
move-panes 0.2.0 
multi-cursor 2.1.5 
open-recent 5.0.0 
open-terminal-here 2.3.1 
pigments 0.40.2 
platformio-ide-terminal 2.7.0 
postcss-sorting 2.10.0 
project-manager 3.3.5 
regex-railroad-diagram 0.19.4 
stylefmt 5.0.1 
Sublime-Style-Column-Selection 1.7.4 
tablr 1.8.3 
term3 0.22.1 
toggle-quotes 1.1.0 
trailing-spaces 0.4.0 
vim-mode-plus 1.24.3 
vim-mode-visual-block 0.2.16 
wordcount 2.10.4 
```

**Comment by t9md** [comment]
I cannot gey situation, can you explain it with sample code?

**Comment by ysakmrkm** [comment]
It seems like occurred in wrapped line.
in 368 / 376, error occurred.
![screen shot 2017-12-19 at 15 55 45](https://user-images.githubusercontent.com/968001/34144519-2c6462dc-e4d5-11e7-973c-9240071cdb30.png)

**Comment by t9md** [comment]
I cannot understand. Describe reprosteps by referring the GIF you attached.

**Comment by basicBrogrammer** [comment]
I think it happens going over a fold with soft warp turned on. 
![image](https://user-images.githubusercontent.com/6913826/34170340-8496285e-e4b8-11e7-82bf-0a7c5277c049.png)

When I maximize my pane, it doesn't throw the error.

**Comment by t9md** [comment]
OK, I can reproduce it now, thanks @basicBrogrammer .

**Comment by t9md** [comment]
Released v1.24.4 @ysakmrkm thanks for reporting.
